modules/snmp-dest/snmpdest.c

```diff
@@ -705,78 +623,38 @@ static gboolean
 snmpdest_dd_init(LogPipe *s)
 {
   SNMPDestDriver *self = (SNMPDestDriver *)s;
+  GlobalConfig *cfg = log_pipe_get_config(s);
 
-  if (!log_dest_driver_init_method(s))
+  if (!log_threaded_dest_driver_init_method(s))
     return FALSE;
 
-  /* prepare the host:port format - the session.remote_port doesn't work */
-  self->host_port = g_string_sized_new(64);
-  if (self->transport)
-    g_string_append_printf(self->host_port, "%s:", self->transport);
-  g_string_append_printf(self->host_port, "%s:%d", self->host, self->port);
-
-  self->queue = log_dest_driver_acquire_queue(&self->super, snmpdest_dd_format_persist_name(self));
-
-  guint SCS_SNMP = stats_register_type("snmp");
-
-  stats_lock();
-  stats_cluster_logpipe_key_set(&self->sc_key_queued, SCS_SNMP | SCS_DESTINATION,
```

Comment:
I think it is worth to mention, that statistics output will change with this change.

1. `LogThreadedDestDriver` registers the following statistics:
```
SC_TYPE_DROPPED
SC_TYPE_PROCESSED
SC_TYPE_WRITTEN
```
(and we got `SC_TYPE_QUEUED` via logqueue)

2. I tried to test the original statistics on the latest master. They did not work for me.
 Thus I would not consider this change as an "error" (quite the opposite), but I will mark it as "user visible"

Reply:
I created a ticket for the techwriters to document the different behavior.